components/script/dom/element.rs

```diff
@@ -3292,7 +3295,30 @@ impl Element {
         {
             return rect;
         }
-        let rect = self.upcast::<Node>().client_rect();
+
+        let mut rect = self.upcast::<Node>().client_rect();
+        let owner_doc = self.node.owner_doc();
+
+        if (matches!(owner_doc.quirks_mode(), QuirksMode::Quirks) &&
+            owner_doc.GetBody().as_deref() == self.downcast::<HTMLElement>()) ||
+            (matches!(owner_doc.quirks_mode(), QuirksMode::NoQuirks) &&
+                *self.root_element() == *self)
+        {
+            let viewport = owner_doc.window().current_viewport();
+
+            //cast Rect<Au, _> into Rect<i32, _>
+            rect = Rect::<i32>::new(
+                Point2D::<i32>::new(
+                    viewport.origin.x.to_nearest_px(),
+                    viewport.origin.y.to_nearest_px(),
+                ),
+                Size2D::<i32>::new(
+                    viewport.size.width.to_nearest_px(),
+                    viewport.size.height.to_nearest_px(),
+                ),
+            );
```

Comment:
If you look at how `current_viewport` is used in window.rs, you can see it is tracking the current scrollport position in the document. I don't think that's the value you want here. Instead you can simply modify the width and height here:

```rust
rect.size = Size2D::<i32>::new(
...using self.initial_viewport...
);
```